### PR TITLE
Remove protobuf dependency for config.Config object

### PR DESCRIPTION
**Pull request by howardjohn**
The problem we face is the current Config object requires the spec to be a (gogo) protobuf. This is problematic for a number of reasons:

* We may change our protos to golang protobuf
* Projects we want to import do not have any protos
* Incorrect abstraction

We don't actually need protobuf - what we need is a way to marshal, unmarshal, and deepcopy any type we want to reference in Config.

### CODE REVIEW

File: pilot/test/mock/config.go [hunk 1]
```diff
@@ -294,7 +293,7 @@ func CheckIstioConfigTypes(store model.ConfigStore, namespace string, t *testing
 		name       string
 		configName string
 		schema     collection.Schema
-		spec       proto.Message
+		spec       config2.Spec
```
Comment (nmittler) [inline pilot/test/mock/config.go:296]: nit: better package name?

File: pkg/config/schema/resource/schema.go [hunk 2]
```diff
@@ -99,6 +100,9 @@ type Builder struct {
 	// Proto refers to the protobuf message type name corresponding to the type
 	Proto string
 
+	// ReflectType is the type of the go struct
+	ReflectType reflect.Type
```
Comment (nmittler) [inline pkg/config/schema/resource/schema.go:104]: Maybe update the comments here and for `Proto`/`ProtoPackage` to make it clear how they work together.

**Review by nmittler — APPROVED** [review]

**Comment by howardjohn** [comment]
@rshriram can you take a look?

**Review by rshriram — APPROVED** [review]

**Pull request merged**